src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -16525,9 +16525,23 @@ void Unit::SetContestedPvP(Player* attackedPlayer)
 {
     Player* player = GetCharmerOrOwnerPlayerOrPlayerItself();
 
-    if (!player || (attackedPlayer && (attackedPlayer == player || (player->duel && player->duel->opponent == attackedPlayer))))
+    if (!player || ((attackedPlayer && (attackedPlayer == player || (player->duel && player->duel->opponent == attackedPlayer))) || player->InBattleground()))
         return;
 
+    // check if there any any guards that should give a fuck about the contested flag on player
+
+    std::list<Unit*>                                                                targets;
+    Acore::NearestVisibleDetectableContestedGuardUnitCheck                          u_check(this);
+    Acore::UnitListSearcher<Acore::NearestVisibleDetectableContestedGuardUnitCheck> searcher(this, targets, u_check);
+    Cell::VisitAllObjects(this, searcher, MAX_AGGRO_RADIUS);
+
+    // return if there are no contested guards found
+
```

Comment:
connect the comment to right above a line of code so it is not just hanging in the air =)